lib/saito/ui/game-playerbox/game-playerbox.js

```diff
@@ -1,85 +1,147 @@
+ *   Graphic can be assigned classnames for more flexible display behavior (such as outside the player-box)
+ */
 class GamePlayerBox {
 
+    /**
+     *  @constructor 
+     *  @param app - Saito app
+     */
     constructor(app) {
       this.app = app;
       this.game_mod = null;
     }
 
 
-    /*
-      Test is DOM contains element ID player-info-1, if not 
-      create all the player boxes, positioned by PlayersBoxArray 
+    /**
+     * Creates player boxes according to number of players in game
+     * Automatically assigns unique id #s to divs in DOM, but can be specified by setting a seats property in game object
+     * @param app - Saito app
+     * @param game_mod - the game object
     */
     render(app, game_mod) {
       this.game_mod = game_mod;
-      
-      try {
-        let playerboxes = [];
-         
+      try { 
         for (let i = 1; i <= game_mod.game.players.length; i++) {
           let player = this._playerBox(i);
-          if (!document.getElementById("player-box-"+player)) {
+          if (!document.getElementById(`player-box-${player}`)) {
+            //let isItMe = (this.returnPlayersBoxArray().indexOf(player) == 0); //Include status in box of the controlling player
             let pBox = app.browser.htmlToElement(GamePlayerBoxTemplate(player));
             if (document.querySelector(".main")) {
               document.querySelector(".main").append(pBox);
             } else { 
-	      document.body.append(pBox);
-	    }
+	            document.body.append(pBox);
+	          }
             this.refreshName(i); //Player names included in box by default
           }
-          if (document.getElementById(("player-box-"+player))) {
-            document.getElementById(("player-box-"+player)).style.display = "block";
+          if (document.getElementById((`player-box-${player}`))) {
+            document.getElementById((`player-box-${player}`)).style.display = "block";
           }
         }
       } catch (err) {console.log("Render error: ",err);}
     }
 
 
+    /** Doesn't do anything */
+    attachEvents(app, data=null) {
+        let chatmod = null;
+        let pb_self = this;
+          for (let i = 0; i < this.app.modules.mods.length; i++){
+            if (this.app.modules.mods[i].slug === "chat") {
+              chatmod = this.app.modules.mods[i];
+            }
+          } 
+      
+        for (let player = 1; player <= this.game_mod.game.players.length; player++) {
+           let boxId = this._playerBox(player);
+           $(`#player-box-head-${boxId}`).off();
+           $(`#player-box-head-${boxId}`).on("dblclick",function(){
+            //Code to launch a private chat window
+            let members = [pb_self.game_mod.game.players[player-1], pb_self.app.wallet.returnPublicKey()].sort();
+            let gid = pb_self.app.crypto.hash(members.join('_'));
+            let newgroup = chatmod.createChatGroup(members, `Player ${player}`);
+              if (newgroup) {
```

Comment:
Wanna paste touched jf-files into https://prettier.io/ to get the format right?

Reply:
I love the brackets, Dan.

Bookmarked link Hannes.